File: app/src/main/java/com/example/ui/components/DebtSummaryCard.kt
package com.example.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.automirrored.filled.ArrowForward
import androidx.compose.material.icons.filled.AccountBalance
import androidx.compose.material.icons.filled.AccountBalanceWallet
import androidx.compose.material.icons.filled.ArrowDownward
import androidx.compose.material.icons.filled.ArrowUpward
import androidx.compose.material.icons.filled.People
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.core.theme.DesignTokens
import com.example.core.util.CurrencyFormatter
import com.example.ui.theme.GreenIncome
import com.example.ui.theme.RedExpense

@Composable
fun DebtSummaryCard(
    totalReceivables: Double,
    totalPayables: Double,
    currency: String = "SAR",
    personsCount: Int = 0,
    ledgersCount: Int = 0,
    isArabic: Boolean = false,
    modifier: Modifier = Modifier
) {
    val netDebt = totalReceivables - totalPayables
    val isNetOwed = netDebt < 0

    Card(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 4.dp),
        shape = RoundedCornerShape(20.dp),
        colors = CardDefaults.cardColors(containerColor = Color(0xFFF3EFF7)),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 16.dp, vertical = 10.dp)
        ) {
            // Top Row: Badge pill on one side, Label + Wallet Icon on the other side
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                // Top Badge Pill
                Surface(
                    shape = RoundedCornerShape(50.dp),
                    color = if (isNetOwed) Color(0xFFFCE8E6) else Color(0xFFE6F4EA)
                ) {
                    Text(
                        text = if (isNetOwed) {
                            if (isArabic) "(-) صافي عليك" else "(-) Net Owed"
                        } else {
                            if (isArabic) "(+) صافي لك" else "(+) Net Asset"
                        },
                        style = MaterialTheme.typography.labelMedium.copy(
                            fontWeight = FontWeight.Bold,
                            color = if (isNetOwed) Color(0xFFB91C1C) else Color(0xFF15803D),
                            fontSize = 12.sp
                        ),
                        modifier = Modifier.padding(horizontal = 10.dp, vertical = 4.dp)
                    )
                }

                // Title + Purple Circle Wallet Icon
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Text(
                        text = if (isArabic) "صافي رصيد الديون" else "Net Debt Balance",
                        style = MaterialTheme.typography.labelMedium.copy(
                            fontWeight = FontWeight.SemiBold,
                            color = Color(0xFF64748B),
                            fontSize = 13.sp
                        )
                    )
                    Spacer(modifier = Modifier.width(8.dp))
                    Box(
                        modifier = Modifier
                            .size(32.dp)
                            .clip(CircleShape)
                            .background(Color(0xFFE9D8FD)),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            imageVector = Icons.Default.AccountBalanceWallet,
                            contentDescription = "Wallet",
                            tint = Color(0xFF6B46C1),
                            modifier = Modifier.size(18.dp)
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.height(6.dp))

            // Main Large Net Amount Text
            val formattedAmountStr = CurrencyFormatter.format(kotlin.math.abs(netDebt), currency) + (if (isNetOwed) "-" else "+")
            Text(
                text = formattedAmountStr,
                style = MaterialTheme.typography.headlineLarge.copy(
                    fontWeight = FontWeight.ExtraBold,
                    fontSize = 30.sp,
                    color = if (isNetOwed) Color(0xFFB91C1C) else Color(0xFF15803D)
                ),
                modifier = Modifier.align(Alignment.End)
            )

            Spacer(modifier = Modifier.height(10.dp))

            // Bottom Sub-cards Row (Payables & Receivables) in White Cards
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(10.dp)
            ) {
                // Payables White Card
                Card(
                    modifier = Modifier.weight(1f),
                    shape = RoundedCornerShape(14.dp),
                    colors = CardDefaults.cardColors(containerColor = Color.White),
                    elevation = CardDefaults.cardElevation(defaultElevation = 1.dp)
                ) {
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(horizontal = 12.dp, vertical = 8.dp),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column {
                            Text(
                                text = if (isArabic) "التزامات (عليه)" else "Payables",
                                style = MaterialTheme.typography.labelSmall.copy(
                                    color = Color(0xFF64748B),
                                    fontSize = 11.sp,
                                    fontWeight = FontWeight.Medium
                                )
                            )
                            Spacer(modifier = Modifier.height(2.dp))
                            Text(
                                text = CurrencyFormatter.format(totalPayables, currency),
                                style = MaterialTheme.typography.titleMedium.copy(
                                    fontWeight = FontWeight.Bold,
                                    color = Color(0xFFB91C1C),
                                    fontSize = 14.sp
                                )
                            )
                        }

                        Box(
                            modifier = Modifier
                                .size(28.dp)
                                .clip(CircleShape)
                                .background(Color(0xFFFCE8E6)),
                            contentAlignment = Alignment.Center
                        ) {
                            Icon(
                                imageVector = Icons.Default.ArrowUpward,
                                contentDescription = "Payables",
                                tint = Color(0xFFB91C1C),
                                modifier = Modifier.size(16.dp)
                            )
                        }
                    }
                }

                // Receivables White Card
                Card(
                    modifier = Modifier.weight(1f),
                    shape = RoundedCornerShape(14.dp),
                    colors = CardDefaults.cardColors(containerColor = Color.White),
                    elevation = CardDefaults.cardElevation(defaultElevation = 1.dp)
                ) {
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(horizontal = 12.dp, vertical = 8.dp),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column {
                            Text(
                                text = if (isArabic) "مستحقات (له)" else "Receivables",
                                style = MaterialTheme.typography.labelSmall.copy(
                                    color = Color(0xFF64748B),
                                    fontSize = 11.sp,
                                    fontWeight = FontWeight.Medium
                                )
                            )
                            Spacer(modifier = Modifier.height(2.dp))
                            Text(
                                text = CurrencyFormatter.format(totalReceivables, currency),
                                style = MaterialTheme.typography.titleMedium.copy(
                                    fontWeight = FontWeight.Bold,
                                    color = Color(0xFF15803D),
                                    fontSize = 14.sp
                                )
                            )
                        }

                        Box(
                            modifier = Modifier
                                .size(28.dp)
                                .clip(CircleShape)
                                .background(Color(0xFFE6F4EA)),
                            contentAlignment = Alignment.Center
                        ) {
                            Icon(
                                imageVector = Icons.Default.ArrowDownward,
                                contentDescription = "Receivables",
                                tint = Color(0xFF15803D),
                                modifier = Modifier.size(16.dp)
                            )
                        }
                    }
                }
            }

            // Bottom metadata counts
            if (personsCount > 0 || ledgersCount > 0) {
                Spacer(modifier = Modifier.height(8.dp))
                HorizontalDivider(color = Color(0xFFE2E8F0))
                Spacer(modifier = Modifier.height(6.dp))

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            imageVector = Icons.Default.People,
                            contentDescription = null,
                            tint = Color(0xFF6B46C1),
                            modifier = Modifier.size(15.dp)
                        )
                        Spacer(modifier = Modifier.width(6.dp))
                        Text(
                            text = if (isArabic) "عدد الأشخاص: $personsCount" else "Persons: $personsCount",
                            style = MaterialTheme.typography.labelMedium.copy(
                                fontWeight = FontWeight.SemiBold,
                                color = Color(0xFF64748B),
                                fontSize = 11.sp
                            )
                        )
                    }

                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            imageVector = Icons.Default.AccountBalance,
                            contentDescription = null,
                            tint = Color(0xFF6B46C1),
                            modifier = Modifier.size(15.dp)
                        )
                        Spacer(modifier = Modifier.width(6.dp))
                        Text(
                            text = if (isArabic) "حسابات الديون: $ledgersCount" else "Ledgers: $ledgersCount",
                            style = MaterialTheme.typography.labelMedium.copy(
                                fontWeight = FontWeight.SemiBold,
                                color = Color(0xFF64748B),
                                fontSize = 11.sp
                            )
                        )
                    }
                }
            }
        }
    }
}

File: lib/ui/components/debt_summary_card.dart
import 'package:flutter/material.dart';
import '../../core/theme/app_colors.dart';
import '../../core/theme/design_tokens.dart';
import '../../core/util/currency_formatter.dart';

class DebtSummaryCard extends StatelessWidget {
  final double totalReceivables;
  final double totalPayables;
  final bool isArabic;

  const DebtSummaryCard({
    super.key,
    required this.totalReceivables,
    required this.totalPayables,
    required this.isArabic,
  });

  @override
  Widget build(BuildContext context) {
    final netDebt = totalReceivables - totalPayables;

    return Card(
      margin: const EdgeInsets.all(DesignTokens.spaceMedium),
      color: AppColors.slateDarkCard,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(DesignTokens.radiusLarge),
        side: const BorderSide(color: AppColors.slateBorder),
      ),
      child: Padding(
        padding: const EdgeInsets.all(DesignTokens.spaceMediumLarge),
        child: Column(
          children: [
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Row(
                        children: [
                          const Icon(Icons.arrow_downward, color: AppColors.greenIncome, size: 16),
                          const SizedBox(width: 4),
                          Text(isArabic ? "ديون لك (مستحقات)" : "Receivables", style: const TextStyle(fontSize: 12, color: AppColors.textSecondaryDark)),
                        ],
                      ),
                      const SizedBox(height: 4),
                      Text(
                        CurrencyFormatter.format(totalReceivables, "SAR"),
                        style: const TextStyle(fontSize: 18, fontWeight: FontWeight.bold, color: AppColors.greenIncome),
                      ),
                    ],
                  ),
                ),
                Container(height: 40, width: 1, color: AppColors.slateBorder),
                Expanded(
                  child: Padding(
                    padding: const EdgeInsets.only(left: DesignTokens.spaceMedium),
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Row(
                          children: [
                            const Icon(Icons.arrow_upward, color: AppColors.redExpense, size: 16),
                            const SizedBox(width: 4),
                            Text(isArabic ? "ديون عليك (التزامات)" : "Payables", style: const TextStyle(fontSize: 12, color: AppColors.textSecondaryDark)),
                          ],
                        ),
                        const SizedBox(height: 4),
                        Text(
                          CurrencyFormatter.format(totalPayables, "SAR"),
                          style: const TextStyle(fontSize: 18, fontWeight: FontWeight.bold, color: AppColors.redExpense),
                        ),
                      ],
                    ),
                  ),
                ),
              ],
            ),
            const SizedBox(height: DesignTokens.spaceMedium),
            Container(
              padding: const EdgeInsets.all(DesignTokens.spaceSmall),
              decoration: BoxDecoration(
                color: AppColors.slateDarkBackground,
                borderRadius: BorderRadius.circular(DesignTokens.radiusSmall),
              ),
              child: Row(
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: [
                  Text(isArabic ? "صافي الديون" : "Net Debt Position", style: const TextStyle(fontSize: 12, color: AppColors.textSecondaryDark)),
                  Text(
                    CurrencyFormatter.format(netDebt, "SAR"),
                    style: TextStyle(
                      fontWeight: FontWeight.bold,
                      color: netDebt >= 0 ? AppColors.greenIncome : AppColors.redExpense,
                    ),
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }
}
